### PR TITLE
Export jstring_ and scientific parsers

### PR DESCRIPTION
**Pull request by phadej**
This will let me play with `TokenStream` stuff independently of `aeson`.

### CODE REVIEW

**Comment by bergmark** [comment]
Thanks!

If we consider these "internal", should we really export them from `Data.Aeson.Parser`?

**Comment by phadej** [comment]
@bergmark `Parser.Internal` is not exported module.

**Comment by bergmark** [comment]
Yeah, I meant to export Parser.Internal instead, sorry for not being explicit.

Since we already have the `Internal` convention of things we don't version I'd vote to keep that even if the name doesn't really match, to avoid confusion. `jsstring_` would be weird to have in the public API both based on the naming and implementation O_o. What do you think?

**Comment by phadej** [comment]
I think
- `scientific` would be great to have in a public (versioned) API
- `jstring` (only one `s`!) can stay only in `Parser.Internal`, which is exposed.

**Comment by bergmark** [comment]
Sounds like a good plan!

**Comment by phadej** [comment]
amended the commit

File: Data/Aeson/Parser.hs [hunk 1]
```diff
@@ -46,7 +47,7 @@ module Data.Aeson.Parser
 
 import Prelude ()
 
-import Data.Aeson.Parser.Internal (decodeStrictWith, decodeWith, eitherDecodeStrictWith, eitherDecodeWith, json, json', jstring, value, value')
+import Data.Aeson.Parser.Internal (decodeStrictWith, decodeWith, eitherDecodeStrictWith, eitherDecodeWith, json, json', jstring, jstring_, scientific, value, value')
```
Comment (bergmark) [inline Data/Aeson/Parser.hs]: This probably gives a warning now that jstring_ is unused?

**Comment by bergmark** [comment]
Cheers!

**Pull request merged**